go/tools/fetch_repo/main.go

```diff
@@ -19,19 +19,45 @@ import (
 )
 
 var (
-	remote = flag.String("remote", "", "Go importpath to the repository fetch")
-	rev    = flag.String("rev", "", "target revision")
-	dest   = flag.String("dest", "", "destination directory")
+	remote     = flag.String("remote", "", "The URI of the remote repository. Must be used with the --vcs flag.")
+	cmd        = flag.String("vcs", "", "Version control system to use to fetch the repository. Should be one of: git,hg,svn,bzr. Must be used with the --remote flag.")
+	rev        = flag.String("rev", "", "target revision")
+	dest       = flag.String("dest", "", "destination directory")
+	importpath = flag.String("importpath", "", "Go importpath to the repository fetch")
 )
 
+func getRepoRoot(remote, cmd, importpath string) (*vcs.RepoRoot, error) {
+	r := &vcs.RepoRoot{
+		VCS:  vcs.ByCmd(cmd),
```

Comment:
Check that `VCS` is not nil. Otherwise, we will get a panic later.

Also, I think this should part of the branch below. This assignment shouldn't dominate the call to `RepoRootForImportPath` where we overwrite it.

```go
if cmd != "" && remote != "" {
  r = &vcs.RepoRoot{...}
} else {
  r, err := vcs.RepoRootForImportPath(importpath, true)
  ...
}
```